datumaro/cli/contexts/model.py

```diff
@@ -73,7 +73,7 @@ def add_command(args):
         model_dir = osp.join(project.config.project_dir,
             project.local_model_dir(args.name))
         os.makedirs(model_dir, exist_ok=False)
-        on_error.do(shutil.rmtree, model_dir, ignore_errors=True)
+        Rollback.current().do(shutil.rmtree, model_dir, ignore_errors=True)
```

Comment:
Looks good. Do you have any ideas how can we make it more terse? Probably, I'm OK with the current solution, but I'd prefer calling a single function. Maybe, we could change class name to lower case and join `current().do()` to a single function to make it clear

Reply:
What if I just add something like this to `datumaro.util`:

```python
def on_error_do(...):
    Rollback.current().add(...)
```

Then the usage can be as concise as possible.

> Apparently, in 3.7 we could create a module with an attribute: https://stackoverflow.com/a/57263518

Not a fan of this approach. If you implement it like that, things like autocomplete or go-to-definition won't work anymore, and they are half the reason I'm even making this change in the first place.

Reply:
> `def on_error_do(...):`

Maybe. It's a bit narrower than other ways (in meaning of API accessibility), but it covers all the actual use cases.

Reply:
OK, I implemented it.